### ISSUE TITLE
TS files are preferred/treated as "media" from 3rd-party modules with both JS and TS files

### ISSUE THREAD

**Issue by trxcllnt** [wmonk/create-react-app-typescript#168]
The maintainers of [IxJS](https://github.com/ReactiveX/IxJS) (the iterable/async-iterable sister project to [RxJS](https://github.com/ReactiveX/IxJS)) are exploring module publishing strategies that will most optimally integrate with the diverse set of popular IDEs, build tools, and JS environments. Our goals are:

1. Publish a single `ix` module that can be consumed in node (CommonJS + ESM), Webpack/Rollup/Closure Compiler, Electron, @angular/cli, create-react-app, and TS projects in a structure that "just works" for everyone
2. Enable consumers to determine the iterable/async-iterable [down leveling behavior](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html) via their project config
3. Enable full-source VSCode intellisense, so consumers can explore the TS source code, not just the types/compiled JS

So far, the easiest approach we've found is to publish the TS source files, ES5/CommonJS, and ES2015/ESM targets alongside each other with `.ts`, `.js`, and `.mjs` extensions respectively. We've found this works well with VSCode, `tsc`, `@angular/cli`, and `create-react-app`, and would love to integrate with `create-react-app-typescript` as well.

Right now, the order of module file extensions in the webpack/jest configurations prefers `ts` files over `js`, even when the files are in node_modules. Since `ts-loader` doesn't touch files in node_modules, the TS files are sent to the `file-loader`, and end up in the `static/media` directory. 

Testing locally, I was able to compile with IxJS by changing the order extensions are processed in the webpack ([dev](https://github.com/wmonk/create-react-app-typescript/blob/master/packages/react-scripts/config/webpack.config.dev.js#L96)/[prod](https://github.com/wmonk/create-react-app-typescript/blob/master/packages/react-scripts/config/webpack.config.prod.js#L100)) and [jest](https://github.com/wmonk/create-react-app-typescript/blob/master/packages/react-scripts/scripts/utils/createJestConfig.js#L47) configs from 
```js
[
'.web.ts', '.ts', '.web.tsx', '.tsx',
'.web.js', '.js', '.json', '.web.jsx', '.jsx',
]
```
to 
```js
[
'.web.js', '.js', '.json', '.web.jsx', '.jsx',
'.web.ts', '.ts', '.web.tsx', '.tsx',
]
```

Would the project be open to accepting a PR with this change? This may affect existing `create-react-app-typescript` users with mixed TS/JS code in their `appSrc` directory, so I'd understand if this is too difficult a change to make. If that's the case, do the maintainers here have suggestions on what we could put in our Readme as guidance on how to easily integrate with `create-react-app-typescript`?

Thanks,
Paul

**Comment by DorianGrey** [comment]
I don't think this would be a major issue in most cases, at least with the current setup. In its current state, it should simply ignore all `js` files outside of the `src` directory, due to the `oneOf` rule [here](https://github.com/wmonk/create-react-app-typescript/blob/master/packages/react-scripts/config/webpack.config.dev.js#L134) and the exclusion rule for the `file-loader` [here](https://github.com/wmonk/create-react-app-typescript/blob/master/packages/react-scripts/config/webpack.config.dev.js#L223).
Just not that this version ist not yet released.

Is this still an issue with current master?
If it is - PR is welcome.
